### PR TITLE
Unified Spill numPartitionBits

### PR DESCRIPTION
**Pull request by duanmeng**
Deprecate the old `QueryConfig::kDeprecatedJoinSpillPartitionBits`,
and use the unified `QueryConfig::kNumSpillPartitionBits`.

### CODE REVIEW

**Comment by duanmeng** [comment]
@mbasmanova Hi Masha, could you please help review this PR, I extracted this from the recursive spill PR to make that PR more focused and concise. Thanks a lot.

File: velox/exec/tests/RowNumberTest.cpp [hunk 1]
```diff
@@ -223,6 +193,75 @@ TEST_F(RowNumberTest, largeInput) {
...
+
+  struct {
+    uint32_t spillPartitionBits;
+
+    std::string debugString() const {
+      return fmt::format("SpillPartitionBits {}", spillPartitionBits);
+    }
+  } testSettings[] = {{2}, {3}};
+
+  for (const auto& testData : testSettings) {
```
Comment (xiaoxmeng) [inline velox/exec/tests/RowNumberTest.cpp:214]: Can we just use test injection for this with different partition bits? Thanks!

File: velox/exec/tests/RowNumberTest.cpp [hunk 2]
```diff
@@ -247,7 +286,20 @@ TEST_F(RowNumberTest, maxSpillBytes) {
   for (const auto& testData : testSettings) {
     SCOPED_TRACE(testData.debugString());
     try {
-      TestScopedSpillInjection scopedSpillInjection(100);
+      std::atomic_int numRound{0};
```
Comment (xiaoxmeng) [inline velox/exec/tests/RowNumberTest.cpp:289]: ditto

File: velox/exec/tests/RowNumberTest.cpp [hunk 3]
```diff
@@ -288,8 +340,22 @@ TEST_F(RowNumberTest, memoryUsage) {
     auto spillDirectory = exec::test::TempDirectoryPath::create();
     const std::string spillEnableConfig = std::to_string(spillEnable);
 
+    std::atomic_int numRound{0};
```
Comment (xiaoxmeng) [inline velox/exec/tests/RowNumberTest.cpp:343]: ditto

**Review by xiaoxmeng — COMMENTED** [review]
@duanmeng thanks for the refactor % comments.

File: velox/exec/tests/RowNumberTest.cpp [hunk 4]
```diff
@@ -223,6 +200,55 @@ TEST_F(RowNumberTest, largeInput) {
   testLimit(5'000);
 }
 
+DEBUG_ONLY_TEST_F(RowNumberTest, spill) {
```
Comment (xiaoxmeng) [inline velox/exec/tests/RowNumberTest.cpp:203]:
```
s/DEBUG_ONLY_TEST_F/TEST_F/
```

**Review by xiaoxmeng — APPROVED** [review]
@duanmeng LGTM. Thanks!

File: velox/exec/tests/RowNumberTest.cpp [hunk 5]
```diff
@@ -223,6 +200,55 @@ TEST_F(RowNumberTest, largeInput) {
...
+    TestScopedSpillInjection scopedSpillInjection(100);
+
+    core::PlanNodeId rowNumberPlanNodeId;
+    auto task =
+        AssertQueryBuilder(duckDbQueryRunner_)
+            .spillDirectory(spillDirectory->path)
+            .config(core::QueryConfig::kSpillEnabled, true)
+            .config(core::QueryConfig::kRowNumberSpillEnabled, true)
+            .config(
+                core::QueryConfig::kJoinSpillPartitionBits,
```
Comment (xiaoxmeng) [inline velox/exec/tests/RowNumberTest.cpp:228]: Can you use kRowSpillPartitionBits?

File: velox/common/base/SpillConfig.h [hunk 6]
```diff
@@ -121,11 +120,13 @@ struct SpillConfig {
 
   /// Used to calculate the spill hash partition number for hash join with
   /// 'startPartitionBit'.
+  ///
+  /// DEPRECATED.
   uint8_t joinPartitionBits;
```
Comment (xiaoxmeng) [inline velox/common/base/SpillConfig.h:125]: You can remove here.

File: velox/docs/configs.rst [hunk 7]
```diff
@@ -313,9 +313,14 @@ Spilling
      - integer
      - 29
      - The start partition bit which is used with `spiller_partition_bits` together to calculate the spilling partition number.
-   * - join_spiller_partition_bits
+   * - join_spiller_partition_bits (Deprecated)
```
Comment (xiaoxmeng) [inline velox/docs/configs.rst:316]: Remove here?

**Review by xiaoxmeng — COMMENTED** [review]
@duanmeng thanks for the update!

File: velox/core/QueryConfig.h [hunk 8]
```diff
@@ -270,11 +270,12 @@ class QueryConfig {
   static constexpr const char* kSpillStartPartitionBit =
       "spiller_start_partition_bit";
 
-  static constexpr const char* kJoinSpillPartitionBits =
+  /// DEPRECATED.
+  static constexpr const char* kDeprecatedJoinSpillPartitionBits =
```
Comment (xiaoxmeng) [inline velox/core/QueryConfig.h:274]: This needs to be same
```
/// !!! DEPRECATED: do not use.
```

**Pull request closed without merging**